### DISCUSSION TITLE
Static files url does not support baseurl

### DISCUSSION THREAD

**Discussion by vsoch** [mmistakes/minimal-mistakes#2553] [Q&A]
Hey minimal mistakes maintainers! We are using the theme (and it’s quite lovely) for one of our sites, and [had an issue](https://github.com/DE-RSE/SORSE20/pull/30#issuecomment-640037318) with the following path:

https://github.com/mmistakes/minimal-mistakes/blob/3ab6ba0b1fa35a57d2a729267640e955c35a93e0/_includes/head.html

Specifically, if we customize this during a preview build on circleci the relative url alone (without the site.url nor baseurl) does not suffice to correctly do the change - the only fix we were able to do (using this as a remote template) was to grep for the line in all rendered files in the _site folder and change the entire line to include the customized path.

Would there be any consideration to change this url to use site.url and/or site.baseurl? Thank you kindly!

- Minimal Mistakes version: master

## Expected behavior

Changing site.url and/or site.baseurl renders correct static file paths.

## Steps to reproduce the behavior

Render a site as an artifact on circle with a longer, custom baseurl.

**Comment by iBug** [comment]
Your first link has gone 404. Is it a private repository? If so, could you please copy the relevant details here. I can't figure out what you're talking about right now.

**Comment by vsoch** [comment]
So sorry about that! Yes it's a private repository and this totally slipped my mind! I can give a few direct examples from the build, maybe that would work? (The link was to the PR where we were discussing). here is discussion from the link (maybe not so helpful)

![image](https://user-images.githubusercontent.com/814322/83948735-7d985380-a7dc-11ea-9200-6158473f6f56.png)

And here are examples. This is the base case that we would want to work - changing the baseurl to the circle preview url, and then building the site:

```
baseurl: https://8-267395254-gh.circle-artifacts.com/0/SORSE
url: ''
```
but of course it doesn't render:

![image](https://user-images.githubusercontent.com/814322/83948764-d8ca4600-a7dc-11ea-94ed-db380a563361.png)

and we tried various renditions of that, all times there was a leading slash at the start of the url, the incorrect path, or some derivation of that. So we were wondering instead of using `relative_path` if it might be possible to use `site.url` or `site.baseurl` so the path renders as the others do? And if not, what have you used to build and preview the site on a service like CircleCI? Thanks!

**Comment by iBug** [comment]
https://github.com/mmistakes/minimal-mistakes/blob/5021021fc811ee10e1af5dc8cdfaff4707563a35/_config.yml#L24-L25

**Comment by vsoch** [comment]
We did try that, but perhaps we didn't do the right level of nesting. I'm reading here (I think this is a username associated with your template!) 

![image](https://user-images.githubusercontent.com/814322/83949298-33b16c80-a7e0-11ea-97c0-3800a3c5d835.png)

and it does appear that `relative_url` puts together site.url and site.baseurl. I think perhaps the issue might have been having an empty site.url then, that would determine it to be / and then have a leading slash? OR having https be in the site.baseurl? What should we try instead? Maybe:

```
baseurl: /0/SORSE
url: https://8-267395254-gh.circle-artifacts.com
```
would the additional path before the final baseurl need to be a part of baseurl or url?

**Comment by iBug** [comment]
> What should we try instead? Maybe:
> 
> ```yaml
> baseurl: /0/SORSE
> url: https://8-267395254-gh.circle-artifacts.com
> ```

This is right.

**Comment by vsoch** [comment]
Awesome thanks! I'll give this a try. Thank you for your speedy response and help.

**Comment by mmistakes** [comment]
For reference, this is how `url` and `baseurl` work. These are not unique to the theme and are Jekyll filters. For additional help on these sorts of issues I suggest reaching out to the community on Jekyll's official forum. https://talk.jekyllrb.com/

**Comment by vsoch** [comment]
Thanks! I’m actually a fairly experienced Jekyll user, just had a brain fart it seems :)
